exp/lighthorizon/services/main.go

```diff
@@ -0,0 +1,192 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/stellar/go/exp/lighthorizon/archive"
+	"github.com/stellar/go/exp/lighthorizon/common"
+	"github.com/stellar/go/exp/lighthorizon/index"
+	"github.com/stellar/go/toid"
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+)
+
+type LightHorizon struct {
+	Operations   OperationsService
+	Transactions TransactionsService
+}
+
+type Config struct {
+	Archive    archive.Archive
+	IndexStore index.Store
+	Passphrase string
+}
+
+type TransactionsService struct {
+	TransactionRepository
+	Config Config
+}
+
+type OperationsService struct {
+	OperationsRepository,
+	Config Config
+}
+
+type OperationsRepository interface {
+	GetOperationsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Operation, error)
+}
+
+type TransactionRepository interface {
+	GetTransactionsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Transaction, error)
+}
+
+type searchCallback func(archive.LedgerTransaction, *xdr.LedgerHeader) (finished bool, err error)
+
+func (os *OperationsService) GetOperationsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Operation, error) {
+	ops := []common.Operation{}
+	opsCallback := func(tx archive.LedgerTransaction, ledgerHeader *xdr.LedgerHeader) (bool, error) {
+		for operationOrder, op := range tx.Envelope.Operations() {
+			opParticipants, opParticipantErr := os.Config.Archive.GetOperationParticipants(tx, op, operationOrder+1)
+			if opParticipantErr != nil {
+				return false, opParticipantErr
+			}
+			if _, foundInOp := opParticipants[accountId]; foundInOp {
+				ops = append(ops, common.Operation{
+					TransactionEnvelope: &tx.Envelope,
+					TransactionResult:   &tx.Result.Result,
+					LedgerHeader:        ledgerHeader,
+					TxIndex:             int32(tx.Index),
+					OpIndex:             int32(operationOrder + 1),
+				})
+				if int64(len(ops)) == limit {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	}
+
+	if err := searchTxByAccount(ctx, cursor, accountId, os.Config, opsCallback); err != nil {
+		return nil, err
+	}
+
+	return ops, nil
+}
+
+func (ts *TransactionsService) GetTransactionsByAccount(ctx context.Context, cursor int64, limit int64, accountId string) ([]common.Transaction, error) {
+	txs := []common.Transaction{}
+
+	txsCallback := func(tx archive.LedgerTransaction, ledgerHeader *xdr.LedgerHeader) (bool, error) {
+		txs = append(txs, common.Transaction{
+			TransactionEnvelope: &tx.Envelope,
+			TransactionResult:   &tx.Result.Result,
+			LedgerHeader:        ledgerHeader,
+			TxIndex:             int32(tx.Index),
+		})
+		if int64(len(txs)) == limit {
+			return true, nil
+		}
+		return false, nil
```

Comment:
suggestion: could combine these :man_shrugging: 
```suggestion
		return (int64(len(txs)) >= limit), nil
```